iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -320,11 +320,8 @@ + (void)setMSubscriptionStatus:(NSNumber*)status {
     mSubscriptionStatus = [status intValue];
 }
 
-static OSDevice* _userDevice;
 + (OSDevice *)getUserDevice {
-    if (!_userDevice)
-        _userDevice = [OSDevice new];
-    return _userDevice;
+    return [[OSDevice alloc] initWithSubscriptionState:[OneSignal getPermissionSubscriptionState]];
```

Comment:
I imagined we will still have this inited once and have getter internally on OSDevice accessing permissionStatus and  subscriptionStatus, what do you think? May I be missing something?

Reply:
If we only init once then there is no change from the previous behavior. The OSDevice will still be the same object that will have the state changing underneath it